Reference for stage 3 temporal (#37344)
mdn/content: files/en-us/web/javascript/reference/global_objects/temporal/now/index.md

---
title: Temporal.Now
slug: Web/JavaScript/Reference/Global_Objects/Temporal/Now
page-type: javascript-namespace
browser-compat: javascript.builtins.Temporal.Now
---

{{JSRef}}

The **`Temporal.Now`** object provides methods for getting the current time in various formats.

## Description

Unlike most global objects, `Temporal.Now` is not a constructor. You cannot use it with the [`new` operator](/en-US/docs/Web/JavaScript/Reference/Operators/new) or invoke the `Temporal.Now` object as a function. All properties and methods of `Temporal.Now` are static (just like the {{jsxref("Math")}} object).

Most fundamentally, the system time is returned by the operating system as a time since the Unix epoch (usually millisecond-level precision, but might be nanosecond-level too). {{jsxref("Temporal/Now/instant", "Temporal.Now.instant()")}} returns this time as a {{jsxref("Temporal.Instant")}} object.

> Then, this instant can further be interpreted in a time zone (which is the system time zone {{jsxref("Temporal/Now/timeZoneId", "Temporal.Now.timeZoneId()")}} by default) in the same fashion as {{jsxref("Temporal/Instant/toZonedDateTimeISO", "Temporal.Instant.prototype.toZonedDateTimeISO()")}}. {{jsxref("Temporal/Now/zonedDateTimeISO", "Temporal.Now.zonedDateTimeISO()")}} returns this interpretation as a {{jsxref("Temporal.ZonedDateTime")}} object. Then, you can get different parts of the date and time, such as the date, time, or date and time together, using {{jsxref("Temporal/Now/plainDateISO", "Temporal.Now.plainDateISO()")}}, {{jsxref("Temporal/Now/plainTimeISO", "Temporal.Now.plainTimeISO()")}}, and {{jsxref("Temporal/Now/plainDateTimeISO", "Temporal.Now.plainDateTimeISO()")}}, respectively.

Reviewer: Maybe you'd like to reword for clarity that you don't need to 'first create an instant and then interpret that in a time zone' if that makes sense:

```suggestion
An instant can be interpreted in a time zone (which is the system time zone {{jsxref("Temporal/Now/timeZoneId", "Temporal.Now.timeZoneId()")}} by default) in the same fashion as {{jsxref("Temporal/Instant/toZonedDateTimeISO", "Temporal.Instant.prototype.toZonedDateTimeISO()")}}. To get a {{jsxref("Temporal.ZonedDateTime")}} object, you can use {{jsxref("Temporal/Now/zonedDateTimeISO", "Temporal.Now.zonedDateTimeISO()")}}. You can also get different parts of the date and time, for example:

- The date using {{jsxref("Temporal/Now/plainDateISO", "Temporal.Now.plainDateISO()")}}
- The time using {{jsxref("Temporal/Now/plainTimeISO", "Temporal.Now.plainTimeISO()")}}
- The date time using {{jsxref("Temporal/Now/plainDateTimeISO", "Temporal.Now.plainDateTimeISO()")}}.
```

Feel free to reword, but I feel "Then you have XYZ … then you can" gives the impression of a sequential operation.

> For example, if the computer is set to the time zone "America/New_York", then `Temporal.Now.zonedDateTimeISO()` may return a zoned date-time like: `2021-08-01T10:40:12.345-04:00[America/New_York]`. Then, `Temporal.Now.plainTimeISO()` may return the time part of this zoned date-time: `10:40:12.345`. However, if you call `Temporal.Now.plainTimeISO("UTC")`, it will return the time part of the zoned date-time in the UTC time zone: `14:40:12.345`. This is especially useful for cross-system communication where the other end may be expecting the time in a different time zone.

Reviewer: Same rationale as above with "then, X":

```suggestion
For example, if the computer is set to the time zone "America/New_York", `Temporal.Now.zonedDateTimeISO()` returns a zoned date-time like: `2021-08-01T10:40:12.345-04:00[America/New_York]`. In this case, `Temporal.Now.plainTimeISO()` would have returned the time part of this zoned date-time: `10:40:12.345`. However, if you call `Temporal.Now.plainTimeISO("UTC")`, it returns the time part of the zoned date-time in the UTC time zone: `14:40:12.345`. This is especially useful for cross-system communication where the other end may be expecting the time in a different time zone.
```

### Reduced time precision

> To offer protection against timing attacks and [fingerprinting](/en-US/docs/Glossary/Fingerprinting), the precision of the `Temporal.Now()` functions might get rounded depending on browser settings. In Firefox, the `privacy.reduceTimerPrecision` preference is enabled by default and defaults to 2ms. You can also enable `privacy.resistFingerprinting`, in which case the precision will be 100ms or the value of `privacy.resistFingerprinting.reduceTimerPrecision.microseconds`, whichever is larger.

Reviewer: We're usually not adding the browser-specifics into prose like this, worth considering with the prefs if they should be in BCD instead, although this feels like important info to surface.

Author: We have a section like this for every single API that deals with system time, and this section seems to have been there since around FF 80.

Reviewer: OK, not blocking, something for us (me?) to note for a follow-up
